cmake/compiler_config.cmake

```diff
@@ -0,0 +1,79 @@
+#------------------------------------------------------------------------------
+# compiler flags/definitions
+#------------------------------------------------------------------------------
+set(gnu_clang_flags
+    # we want to be as strict as possible
+    -Wall
+    $<$<BOOL:${BUILD_STRICT_MODE}>:-Werror>
+    # optimization
+    -msse3
+    # disable warnings
+    -Wno-deprecated
+    -Wno-deprecated-declarations
+    -Wno-unused-local-typedefs
+)
+
+set(msvc_flags
+    # we want to be as strict as possible
+    /W3
```

Comment:
We currently have to use level 3 (/W3) instead of level 4(/W4) for Visual studio compiler since Pixar's USD core needs to adopt their API library to respect this level. Switching to /W4 is out of commission at the moment.

More on /W4 here:
https://docs.microsoft.com/en-us/cpp/build/reference/compiler-option-warning-level?view=vs-2019